modules/effects/src/effect_sources.ts

```diff
@@ -47,7 +41,21 @@ export class EffectSources extends Subject<any> {
   toActions(): Observable<Action> {
     return this.pipe(
       groupBy(getSourceForInstance),
-      mergeMap(source$ => source$.pipe(groupBy(effectsInstance))),
+      mergeMap(source$ => {
+        return source$.pipe(
+          groupBy(effectsInstance),
+          tap(() => {
+            if (
+              onInitEffects in source$.key &&
+              typeof source$.key[onInitEffects] === 'function'
```

Comment:
I think we should push this logic back into lifecycle_hooks.ts and expose it has `isOnInitEffects(key): key is OnInitEffects` type guard

Reply:
I'll do this in another PR if that's OK.
Currently, all checks on lifecycle hooks are done in a similar way, in the next PR we can refactor all of them to type guards.